osdep/compiler.h

```diff
@@ -25,4 +27,17 @@
 #define MP_ASSERT_UNREACHABLE() (assert(!"unreachable"), abort())
 #endif
 
+#ifndef NDEBUG
```

Comment:
For reference, this is what I was referring to.

https://github.com/mpv-player/mpv/blob/f41805081b2c4aee2b4d4519be1b2dbe62c689a0/common/common.h#L155-L159

You can just rename this to `MP_ENSURE` or whatever and then alias `MP_HANDLE_OOM` to it.

I also agree with the decision of not doing any printing here. These macros should be used in "highly unlikely to happen but check just in case" situations, so it doesn't make too much sense to increase the binary size with unnecessary printing code + string literals.